src/components/order-service/order-service.controller.ts

```diff
@@ -94,7 +96,44 @@ export class OrderServiceController implements angular.IController {
     this.ctrl.selectedPlan = plan;
   }
 
+  public invalidNewProject() {
+    // test whether selectedProject is actually a newProject obj. and if passed name validation
+    return this.isNewProject() && !this.ctrl.selectedProject.validName;
+  }
+
   public provisionService() {
+    if (this.isNewProject()) {
+      // the selectedProject is actually a newProject object
+      let newProjName = this.ctrl.selectedProject.metadata.name;
+      let newProjDisplayName = this.ctrl.selectedProject.metadata.annotations['new-display-name'];
```

Comment:
Can we just use the normal annotation for display name `openshift.io/display-name`? It looks like you use the normal description annotation already.